test/unit/action_tests.rb

```diff
@@ -0,0 +1,338 @@
+          from: [],
+          to: [:test_param, :other_param]
+        )
+      assert_that(subject.date_params).equals([:test_param, :other_param])
+
+      # time_params
+      assert_that { subject.time_params(:test_param, :other_param) }
+        .changes(
+          "subject.much_rails_action_config.time_params.dup",
+          from: [],
+          to: [:test_param, :other_param]
+        )
+      assert_that(subject.time_params).equals([:test_param, :other_param])
+
+      # boolean_params
+      assert_that { subject.boolean_params(:test_param, :other_param) }
+        .changes(
+          "subject.much_rails_action_config.boolean_params.dup",
+          from: [],
+          to: [:test_param, :other_param]
+        )
+      assert_that(subject.boolean_params).equals([:test_param, :other_param])
+
+      # format
+      assert_that { subject.format("VALUE") }
+        .changes(
+          "subject.much_rails_action_config.format.dup",
+          from: nil,
+          to: "VALUE"
+        )
+      assert_that(subject.format).equals("VALUE")
+
+      # on_validation
+      assert_that { subject.on_validation(&block1) }
+        .changes(
+          "subject.much_rails_action_config.on_validation_blocks.dup",
+          from: [],
+          to: [block1]
+        )
+
+      # on_validation_blocks
+      assert_that { subject.on_validation(&block2) }
+        .changes(
+          "subject.on_validation_blocks.dup",
+          from: [block1],
+          to: [block1, block2]
+        )
+
+      # on_call, on_call_block
+      subject.on_call(&block1)
+      assert_that(subject.much_rails_action_config.on_call_block)
+        .equals(block1)
+      assert_that(subject.on_call_block).equals(block1)
+
+      # on_before_call
+      assert_that { subject.on_before_call(&block1) }
+        .changes(
+          "subject.much_rails_action_config.on_before_call_blocks.dup",
+          from: [],
+          to: [block1]
+        )
+
+      # on_before_call_blocks
+      assert_that { subject.on_before_call(&block2) }
+        .changes(
+          "subject.on_before_call_blocks.dup",
+          from: [block1],
+          to: [block1, block2]
+        )
+
+      # on_after_call
+      assert_that { subject.on_after_call(&block1) }
+        .changes(
+          "subject.much_rails_action_config.on_after_call_blocks.dup",
+          from: [],
+          to: [block1]
+        )
+
+      # on_after_call_blocks
+      assert_that { subject.on_after_call(&block2) }
+        .changes(
+          "subject.on_after_call_blocks.dup",
+          from: [block1],
+          to: [block1, block2]
+        )
+    end
+  end
+
+  class InitTests < ReceiverTests
+    desc "when init"
+    subject { receiver_class.new(params: {}) }
+
+    should have_imeths :on_call, :valid_action?, :successful_action?
+  end
+
+  class ReceiverCallMethodTests < UnitTests
+    desc "receiver #call method"
+    subject { receiver_class.new(params: params) }
```

Comment:
I'd combine these two contexts - they are virtually identical in subjects:

```suggestion
  class InitTests < ReceiverTests
    desc "when init"
    subject { receiver_class.new(params: params) }

    should have_imeths :on_call, :valid_action?, :successful_action?
```